src/v/cluster/scheduling/allocation_strategy.cc

```diff
@@ -87,24 +87,30 @@ model::node_id find_best_fit(
         if (it == nodes.end()) {
             continue;
         }
-
-        uint32_t score = std::accumulate(
-          constraints.begin(),
-          constraints.end(),
-          0,
-          [&node = it->second](
-            uint32_t score,
-            const allocation_constraints::soft_constraint_ev_ptr& ev) {
-              const auto current_score = ev->score(*node);
-              vlog(
-                clusterlog.trace,
-                "constraint: {}, node: {}, score: {}",
-                *ev,
-                node->id(),
-                current_score);
-              return score + current_score;
-          });
-
+        /**
+         * Score is normalized so that it is always in range [0, max_score_size]
+         */
+        uint32_t score
+          = std::accumulate(
+              constraints.begin(),
+              constraints.end(),
+              uint32_t{0},
+              [&node = it->second](
+                uint32_t score,
+                const allocation_constraints::soft_constraint_ev_ptr& ev) {
+                  const auto current_score = ev->score(*node);
+                  vlog(
+                    clusterlog.trace,
+                    "constraint: {}, node: {}, score: {}",
+                    *ev,
+                    node->id(),
+                    current_score);
+                  return score + current_score;
+              })
+            / constraints.size();
```

Comment:
all nodes will get the same normalizing coefficient so the end result would be the same, right? so the main difference would be in logging or is there something more to it?

Reply:
this change for now is just for consistency the normalization will be required as soon as we will calculate the fitness for partitions in partition balancer. Then we will need to have comparable results no mater how many constraints are applied to given partition.